public/app/features/dashboard/components/DashboardSettings/VersionsSettings.test.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import { within } from '@testing-library/dom';
+import userEvent from '@testing-library/user-event';
+import { historySrv } from '../VersionHistory/HistorySrv';
+import { VersionsSettings, VERSIONS_FETCH_LIMIT } from './VersionsSettings';
+import { versions } from './__mocks__/versions';
+
+jest.mock('../VersionHistory/HistorySrv');
+
+describe('VersionSettings', () => {
+  const dashboard: any = {
+    id: 74,
+    version: 7,
+    formatDate: jest.fn(() => 'date'),
+    getRelativeTime: jest.fn(() => 'time ago'),
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('renders a header and a loading indicator followed by results in a table', async () => {
+    // @ts-ignore
+    historySrv.getHistoryList.mockResolvedValue(versions);
+    render(<VersionsSettings dashboard={dashboard} />);
+    expect(screen.getByRole('heading', { name: /versions/i })).toBeInTheDocument();
+    expect(screen.queryByText(/fetching history list/i)).toBeInTheDocument();
+    await waitFor(() => expect(screen.getByRole('table')).toBeInTheDocument());
+    const tableBodyRows = within(screen.getAllByRole('rowgroup')[1]).getAllByRole('row');
+    expect(tableBodyRows.length).toBe(versions.length);
+    const firstRow = within(screen.getAllByRole('rowgroup')[1]).getAllByRole('row')[0];
+    expect(within(firstRow).getByText(/latest/i)).toBeInTheDocument();
+    expect(within(screen.getByRole('table')).getAllByText(/latest/i)).toHaveLength(1);
```

Comment:
Just a nit about formatting, think this is easier to follow:

```suggestion
    historySrv.getHistoryList.mockResolvedValue(versions);
    render(<VersionsSettings dashboard={dashboard} />);
    
    expect(screen.getByRole('heading', { name: /versions/i })).toBeInTheDocument();
    expect(screen.queryByText(/fetching history list/i)).toBeInTheDocument();
    
    await waitFor(() => expect(screen.getByRole('table')).toBeInTheDocument());
    const tableBodyRows = within(screen.getAllByRole('rowgroup')[1]).getAllByRole('row');
    
    expect(tableBodyRows.length).toBe(versions.length);
    
    const firstRow = within(screen.getAllByRole('rowgroup')[1]).getAllByRole('row')[0];
    
    expect(within(firstRow).getByText(/latest/i)).toBeInTheDocument();
    expect(within(screen.getByRole('table')).getAllByText(/latest/i)).toHaveLength(1);
```